modepy/quadrature/__init__.py

```diff
@@ -82,3 +82,29 @@ def __init__(self, quad, left, right):
         Quadrature.__init__(self,
                 left + (quad.nodes+1) / 2 * length,
                 quad.weights * half_length)
+
+
+class TensorProductQuadrature(Quadrature):
+    def __init__(self, dims, quad):
```

Comment:
Arg order: see above.

Reply:
This seems like it should stay like this? All the `tensor_product_basis` and `tensor_product_nodes` functions take `(dims, thing_1d)`

Reply:
The way I'd like to get rid of this (and `tensor_product_nodes`) is by making an interface that takes a sequence of 1D nodes/quadratures. For `tensor_product_nodes`, it'll probably have to be a new name (`tensor_product_nodes_of([...])`, `tensor_product_nodes_inhom`?), with the existing function deprecated. `TensorProductQuadrature` is new enough that I wouldn't feel bad about just changing it.

Does that sound like a reasonable plan?

Reply:
Yeah, sounds good. This PR introduces the tensor product quadrature stuff, so that's fine! 

This one can probably be hacked around to keep the same name with `tensor_product_nodes(dims_or_list_of_nodes, nodes_1d=None)`? Not sure if it's worth the trouble though.

`tensor_product_nodes_from`? `make_tensor_product_nodes`?

Reply:
I think I like the same-name solution the best. `isinstance(dims, Number)` is pretty bullet-proof.

Reply:
Updated `tensor_product_nodes` in 368bd74 and `TensorProductQuadrature` in ef01010.

Let me know if it looks OK!

Reply:
Thanks for working on that! Left a comment.